test/unittests.jl

```diff
@@ -288,6 +288,12 @@ end
         uX = X .* Quantity(2, length=2.5, luminosity=0.5)
         @test sum(X) == 0.5 * ustrip(sum(uX))
 
+        @test isapprox(uX, uX, atol=Quantity{T,D}(1e-6, length=2.5, luminosity=0.5))
+        @test_throws DimensionError isapprox(uX, uX, atol=1e-6)
+        @test_throws ErrorException DynamicQuantities._norm(1.0)
+        VERSION >= v"1.9" &&
+            @test_throws "Please load the `LinearAlgebra.jl` package." DynamicQuantities._norm(1.0)
```

Comment:
Need test with mixed unit array, like

```julia
julia> x = [1.0u"m", 1.2u"s"]
```

which should be approximately equal to

```julia
julia> y = [1.00000001u"m", 1.2u"s"]
```

Reply:
(Although here it should not be possible to provide an `atol`; it should fall back to `rtol`)

Or maybe this doesn't even make sense to check for...?

Reply:
Will remove this if we want to avoid the per-component fallback (which is probably more expensive to check for anyways).

Reply:
I wrote a comment on one of the reviews above describing my thoughts on mixed-unit vectors. This is one of the complications potentially need to address, that any `atol` argument is pretty much guaranteed to produce a `DimensionError`. Really the only way to handle mixed-units is probably to just internally broadcast the operation. We could also try to catch the condition and generate a descriptive error suggesting they use broadcasting or similar.